File: Analisis_data_shopee/shopee/3.py
import pandas as pd
import mysql.connector
import numpy as np
import matplotlib.pyplot as plt

connection = mysql.connector.connect(host='localhost', user='root', password='', database='pembelian_shopee')
query1 = "SELECT*FROM order_detail"
query2 = "SELECT*FROM sku_detail"
df1 = pd.read_sql(query1, connection)
df2 = pd.read_sql(query2, connection)
connection.close()

df = pd.merge(df1, df2, left_on='sku_id', right_on='id', how='inner')
df = df[df['is_valid'] == 1]
df['order_date'] = pd.to_datetime(df['order_date'])
tabel2021 = df[df['order_date'].dt.year == 2021]
tabel2021 = tabel2021.groupby('category')['after_discount'].sum().round().reset_index(name='2021')
print(tabel2021)
tabel2022 = df[df['order_date'].dt.year == 2022]
tabel2022 = tabel2022.groupby('category')['after_discount'].sum().round().reset_index(name='2022')
print(tabel2022)
hasil = pd.merge(tabel2021, tabel2022, on='category', how='inner')
def pengurangan(x,y):
    return y -x
hasil['perbedaan'] = hasil.apply(lambda row: pengurangan(row['2021'], row['2022']), axis=1)
def Penjelasan(x):
    if x>0:
        return 'Naik'
    elif x<0:
        return 'Turun'
hasil['penjelasan'] = hasil['perbedaan'].apply(Penjelasan)
hasil = hasil.sort_values('perbedaan', ascending=False).reset_index(drop=True)
print(hasil)

File: Analisis_data_shopee/shopee/3.r
library(RMySQL)
library(dplyr)
library(tidyr)
library(lubridate)

# Ganti 'dbname', 'username', 'password', dan 'host' sesuai dengan informasi database Anda
con <- dbConnect(MySQL(), dbname = "pembelian_shopee", user = "root", password = "", host = "localhost")

query1 <- "SELECT * FROM order_detail"
query2 <- "SELECT * FROM sku_detail"

df1 <- dbGetQuery(con, query1)
df2 <- dbGetQuery(con, query2)
dbDisconnect(con)

print(head(df1))
cat("\n")
print(head(df2))
cat("\n\n")

# df <- merge(df1, df2, by.x = "sku_id", by.y = "id", all=FALSE)
df <- inner_join(df1, df2, by = c("sku_id" = "id"), relationship = "many-to-many")
print(head(df))
cat("\n\n")

df <- df[df$is_valid == 1, ]
df$order_date <- as.Date(df$order_date)
print(head(df))



tabel2021 <- df %>% filter(year(order_date) == 2021)
tabel2021 <- tabel2021 %>%
    group_by(category) %>%
    summarise(total2021 = sum(after_discount))
print(head(tabel2021))

cat("\n\n")

tabel2022 <- filter(df, year(order_date) == 2022)
tabel2022 <- tabel2022 %>%
    group_by(category) %>%
    summarise(total2022 = sum(after_discount))
print(head(tabel2022))
cat("\n\n")



hasil <- merge(tabel2021, tabel2022, by = "category", all = FALSE)

pengurangan <- function(x, y) {
    return(y - x)
}
# hasil$perbedaan  <- pengurangan(hasil$total2021, hasil$total2022)
# hasil <- hasil %>% mutate(perbedaan = pengurangan(total2021, total2022))
hasil$perbedaan <- hasil$total2022 - hasil$total2021

penjelasan <- function(x) {
    if (x > 0) {
        return("Naik")
    } else if (x < 0) {
        return("Turun")
    } else {
        return("Tetap")
    }
}
hasil$penjelasan <- sapply(hasil$perbedaan, penjelasan)

hasil <- hasil %>% arrange(desc(perbedaan))
print(hasil)
